### PR TITLE
Add new filters to reservations

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- See ticket for filters.
- Also changes `CustomerTypeChoice`, `ReservationStateChoice`, and `ReservationTypeChoice` enum to uppercase. Refactor some mutations and tests to support this.

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- Requires frontend to implement new enums. Should be pretty simple since the are now GraphQL enums instead of strings.

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3400][TILA]

[TILA]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3400

[TILA-3400]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3400?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by vergama — APPROVED** [review]

File: merchants/models.py [hunk 1]
```diff
@@ -95,6 +95,20 @@ def needs_update_statuses(cls) -> list[OrderStatus]:
         ]
 
 
+class OrderStatusWithFree(models.TextChoices):
+    """Same as OrderStatus, but includes the 'FREE' option used for filtering reservations without payments."""
+
+    # Note: Enums cannot be subclassed, so we have to redefine all "original" members.
```
Comment (ranta) [inline merchants/models.py:101]: Good that this comment is included, as i was about to suggest subclassing the original :D

**Review by ranta — APPROVED** [review]
No tests for the new filters? 🥺
_(It should be fine, they are pretty simple :D)_

**Comment by matti-lamppu** [comment]
Forgot to add most tests (I was busy fixing old ones). Added one for `is_recurring`. Should add quick ones for the rest.

**Comment by matti-lamppu** [comment]
Added tests for the new filters

**Comment by matti-lamppu** [comment]
> Why change some and not all State -> ReservationStateChoice?

It's probably a library issue: Enum name is constructed differently depending on where it is used (unless I do some extra work), and it's not visible to the backend that the name is different. Just point out where the name is different and I can make the change.

**Comment by matti-lamppu** [comment]
The needed enum changes are in the ticket. These should be fixed before mering

**Comment by matti-lamppu** [comment]
Enums should now be correct in the GraphQL schema.

**Review by vergama — APPROVED** [review]

**Pull request merged**